### PR TITLE
chore(nextjs): Improve experience when swapping keys on Keyless mode

### PR DESCRIPTION
**Pull request by panteliselef**
## Description

In Keyless mode it is expected that the user will eventually set keys as environment variables. When they do that, the nextjs server will restart, but there seems to be a mismatch because first the rendering lifecycle runs with the new keys and then the keys are updated in the middleware. This causes an error to be thrown to the user which we can actually improve.

**For the happy path**: we avoid displaying that error by retrying to decrypt data with the dummy key.

**For any other case**: when swapping keys or going back to headless you will now see this (see picture). this does not only reply to keyless.

image

## Checklist

- [ ] `pnpm test` runs as expected.
- [ ] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/nextjs/src/server/utils.ts [hunk 1]
```diff
@@ -280,10 +287,34 @@ export function decryptClerkRequestData(
...
+     *
+     * Attempt one more time with the default dummy value.
+     */
+    if (canUseKeyless__server) {
+      try {
+        return decryptData(encryptedRequestData, KEYLESS_ENCRYPTION_KEY);
+      } catch (e) {
+        throwInvalidEncryptionKey();
+      }
+    }
```
Comment (panteliselef) [inline packages/nextjs/src/server/utils.ts:304]: Simple retrying with the dummy value

**Review by LauraBeatris — APPROVED** [review]

File: .changeset/young-beans-trade.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/nextjs': patch
+---
+
+Improve experience when swapping keys on Keyless mode.
```
Comment (LauraBeatris) [inline .changeset/young-beans-trade.md:5]: 💡 Should we mention here regarding the hot-swapped .env case? Just to clarify the "improve experience" part

File: packages/nextjs/src/server/errors.ts [hunk 3]
```diff
@@ -41,3 +41,5 @@ ${verifyMessage}`;
 export const authSignatureInvalid = `Clerk: Unable to verify request, this usually means the Clerk middleware did not run. Ensure Clerk's middleware is properly integrated and matches the current route. For more information, see: https://clerk.com/docs/references/nextjs/clerk-middleware. (code=auth_signature_invalid)`;
 
 export const encryptionKeyInvalid = `Clerk: Unable to decrypt request data, this usually means the encryption key is invalid. Ensure the encryption key is properly set. For more information, see: https://clerk.com/docs/references/nextjs/clerk-middleware#dynamic-keys. (code=encryption_key_invalid)`;
+
+export const encryptionKeyInvalidDev = `Clerk: Unable to decrypt request data.\n\nRefresh the page if your .env file was just updated. In the issue persists ensure the encryption key is valid and properly set.\n\nFor more information, see: https://clerk.com/docs/references/nextjs/clerk-middleware#dynamic-keys. (code=encryption_key_invalid)`;
```
Comment (BRKalow) [inline packages/nextjs/src/server/errors.ts:45]:
```suggestion
export const encryptionKeyInvalidDev = `Clerk: Unable to decrypt request data.\n\nRefresh the page if your .env file was just updated. If the issue persists, ensure the encryption key is valid and properly set.\n\nFor more information, see: https://clerk.com/docs/references/nextjs/clerk-middleware#dynamic-keys. (code=encryption_key_invalid)`;
```

File: .changeset/young-beans-trade.md (hunk 2 above)
Comment (panteliselef) [inline .changeset/young-beans-trade.md:5]:
```suggestion
Improve error message when swapping keys defined as environment variables. Affects (code=encryption_key_invalid).
 **On Keyless mode**: Attempt to not throw the error.
```

Comment (panteliselef) [reply .changeset/young-beans-trade.md:5]: @BRKalow @LauraBeatris is this better ?

Comment (BRKalow) [inline .changeset/young-beans-trade.md:5]:
```suggestion
Improves the error message when changing keys in development causes a subsequent request to fail.
```

**Pull request merged**